deepchecks/tabular/metric_utils/scorers.py

```diff
@@ -297,20 +298,27 @@ def _run_score(self, model, data: pd.DataFrame, label_col):
             else:
                 label = _transform_to_multi_label_format(np.array(label_col), self.model_classes)
 
+        try:
             scores = self.scorer(updated_model, data, label)
+        except ValueError as e:
+            if getattr(self.scorer, '_score_func', '').__name__ == 'roc_auc_score':
+                get_logger().warning('ROC AUC failed with error message - "%s". setting scores as None', e,
+                                    exc_info=get_logger().level == logging.DEBUG)
+                scores = None
+            else:
+                raise
 
-            # The scores returned are for the observed classes but we want scores of the observed classes
-            if isinstance(scores, t.Sized):
-                if len(scores) != len(self.model_classes):
-                    raise errors.DeepchecksValueError(
-                        f'Scorer returned {len(scores)} scores, but model contains '
-                        f'{len(self.model_classes)} classes. Can\'t proceed')
-                scores = dict(zip(self.model_classes, scores))
-                # Add classes which been seen in the data but are not known to the model
-                scores.update({cls: np.nan for cls in set(self.observed_classes) - set(self.model_classes)})
-            return scores
-        else:
-            return self.scorer(model, data, label_col)
+        # The scores returned are for the observed classes but we want scores of the observed classes
```

Comment:
Can you fix?
`The scores returned are for the observed classes but we want scores of the observed classes` -> `The scores returned are for the observed classes but we want scores of the model classes`